samples: synchronization: updated code and documentation for SMP

youndong opened the pull request:
When samples/synchronization is executed on a board
that supports SMP (eg qemu_x86_64), it shows the execution
results of threads running on different cpu.

Signed-off-by: Youndong Park <youndong.park@gmail.com>

youndong reviewed (commented):
Updated the code and documentation on synchronization sample. README.rst shows the actual results when the binary runs.

stephanosio commented on samples/synchronization/README.rst:
```diff
@@ -30,17 +30,32 @@ Sample Output
-   threadB: Hello World!
-   threadA: Hello World!
-   threadB: Hello World!
+   threed_a: Hello World from cpu(0) on qemu_x86!
```
typo?

stephanosio commented on samples/synchronization/src/main.c:
```diff
@@ -127,6 +127,10 @@ void main(void)
 	k_thread_cpu_mask_enable(&threadB_data, 1);
 #endif
 
+#if CONFIG_SMP
```
```suggestion
#ifdef CONFIG_SMP
```

youndong replied:
Thank you for your correction.

stephanosio reviewed (changes requested):
Please update the commit title to match that of the PR.

youndong committed:
samples: synchronization: updated code and documentation for SMP

When samples/synchronization is executed on a board
that supports SMP (eg qemu_x86_64), it shows the execution
results of threads running on different cpu.

Signed-off-by: Youndong Park <youndong.park@gmail.com>

youndong reviewed (commented):
@stephanosio, I updated some code and README.rst according to your comments. Thank you.

youndong committed:
Merge branch 'zephyrproject-rtos:main' into fix
